2.0 eslint rule newline-per-chained-call bug with array access

```javascript
var currentUrl = window
    .location
    .href
    .match(/(^[^#]*)/)[0];
```
Output:
```
1:18  error  Expected line break after `window\n    .location\n    .href\n    .match(/(^[^#]*)/)`
```

But:
```javascript
var currentUrl = window
    .location
    .href
    .match(/(^[^#]*)/);
```
Work fine.

just an idea to reproduce the issue:

```
17:45:43 [/c/My/Dev]
λ eslint --version
v2.0.0
17:45:55 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)" | eslint --stdin --rule='newline-per-chained-call: 2'
17:46:05 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)[0]" | eslint --stdin --rule='newline-per-chained-call: 2'

<text>
  1:1  error  Expected line break after `window\n.location\n.href\n.match(/(^[^#]*)/)`  newline-per-chained-call

✖ 1 problem (1 error, 0 warnings)

17:46:12 [/c/My/Dev]
λ
```

Be aware, using `ignoreChainWithDepth` (default equals to 2):
http://eslint.org/docs/rules/newline-per-chained-call


```
17:57:56 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)[0]" | eslint --stdin --rule='newline-per-chained-call: [2,{ignor
eChainWithDepth: 4} ]'

<text>
  1:1  error  Expected line break after `window\n.location\n.href\n.match(/(^[^#]*)/)`  newline-per-chained-call

✖ 1 problem (1 error, 0 warnings)

17:58:01 [/c/My/Dev]
λ printf "window\n.location\n.href\n.match(/(^[^#]*)/)[0]" | eslint --stdin --rule='newline-per-chained-call: [2,{ignor
eChainWithDepth: 5} ]'
17:58:05 [/c/My/Dev]
λ
```

Personally , I want to ask if its possible to have an option to chain function calls only and not object access as well?

 I really don't mind the newline-per-chained-call for those but it seems sometimes it gives me a false positive when it shouldn't. Such as what is reported here (Array access) and some weird ones such as `options.level` and `Object.keys(options.meta)`

Example:

```
   37:128  error    Expected line break after `Object.keys(options.meta)`                                          newline-per-chained-call
   37:128  error    Expected line break after `Object`                                                             newline-per-chained-call
   97:19   error    Expected line break after `b[c].split(';')`                                                    newline-per-chained-call
   97:19   error    Expected line break after `b[c]`                                                               newline-per-chained-call
   97:19   error    Expected line break after `b`
```

```
// line 37
                return `${options.timestamp()} ${level} ${(options.message != null ? options.message : '')} ${(options.meta && Object.keys(options.meta).length ? `
                    ${JSON.stringify(options.meta)}
                    ` : '')}`;

// line 97 
        const d = b[c].split(';')[0];
```



Using ESLint Version 2.0 with my [ESLint Config](https://github.com/thedark1337/eslint-config-thedark1337)

--- a/docs/rules/newline-per-chained-call.md
+++ b/docs/rules/newline-per-chained-call.md
@@ -33,7 +33,7 @@ This rule reports such code and encourages new lines after each call in the chai
 
 ## Rule Details
 
-This rule checks and reports the chained calls if there are no new lines after each call or deep member access.
+This rule checks and reports the chained calls if there are no new lines after each call or deep member access. Computed property accesses such as `instance[something]` are excluded.
 
 ## Options
 

--- a/lib/rules/newline-per-chained-call.js
+++ b/lib/rules/newline-per-chained-call.js
@@ -17,7 +17,7 @@ module.exports = function(context) {
         ignoreChainWithDepth = options.ignoreChainWithDepth || 2;
 
     /**
-     * Check and Capture State if the chained calls/memebers
+     * Check and Capture State if the chained calls/members
      * @param {ASTNode} node The node to check
      * @param {object} codeState The state of the code current code to be filled
      * @returns {void}
@@ -36,7 +36,7 @@ module.exports = function(context) {
 
             objectLineNumber = node.object.loc.end.line;
             propertyLineNumber = node.property.loc.end.line;
-            valid = propertyLineNumber > objectLineNumber;
+            valid = node.computed || propertyLineNumber > objectLineNumber;
 
             if (!valid) {
                 codeState.reports.push({
@@ -80,7 +80,7 @@ module.exports = function(context) {
         };
     }
     /**
-     * Process the said node and recuse internally
+     * Process the said node and recurse internally
      * @param {ASTNode} node The node to check
      * @returns {void}
      */

--- a/tests/lib/rules/newline-per-chained-call.js
+++ b/tests/lib/rules/newline-per-chained-call.js
@@ -28,6 +28,10 @@ ruleTester.run("newline-per-chained-call", rule, {
     }, {
         code: "var a = m1();"
     }, {
+        code: "var a = window\n.location\n.href\n.match(/(^[^#]*)/)[0];"
+    }, {
+        code: "var a = window['location']\n.href\n.match(/(^[^#]*)/)[0];"
+    }, {
         code: "var a = m1().m2.m3();",
         options: [{
             ignoreChainWithDepth: 3

